projects/packages/stats/src/class-tracking-pixel.php

```diff
@@ -132,7 +131,7 @@ public static function enqueue_stats_script() {
 		);
 
 		// Make sure the script loads asynchronously (add a defer attribute).
-		Assets::instance()->add_async_script( 'jetpack-stats' );
+		wp_script_add_data( 'jetpack-stats', 'strategy', 'defer' );
```

Comment:
This doesn't seem to work like I would expect:

- It doesn't add a `defer` attribute, just the `data-wp-strategy` attribute set to `defer`.
- Couldn't we update the `wp_enqueue_script` function just above and set it to defer there (although that doesn't seem to add the `defer` attribute either:

```php
wp_enqueue_script(
	'jetpack-stats',
	'https://stats.wp.com/e-' . gmdate( 'YW' ) . '.js',
	array(),
	null, // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion -- The version is set in the URL.
	array(
		'in_footer' => true,
		'strategy'  => 'defer',
	)
);
```

Reply:
> Couldn't we update the wp_enqueue_script function just above and set it to defer there (although that doesn't seem to add the defer attribute either)

Did that in 14dfa23e3154a7fba320cd1a35b739cce1ad0c1a

Reply:
> It doesn't add a `defer` attribute, just the `data-wp-strategy` attribute set to `defer`.

Probably because [WordPress doesn't allow handles with inline "after" scripts to be deferred](https://github.com/WordPress/wordpress-develop/blob/6dd00b1ffac54c20c1c1c7721aeebbcd82d0e378/src/wp-includes/class-wp-scripts.php#L984-L987), which [that handle has](https://github.com/Automattic/jetpack/blob/10a2a12cca11cc4c5025fd11c43d2cfcccbc7d4f/projects/packages/stats/src/class-tracking-pixel.php#L150-L154).